example/github/remote-data-blocks/inc/queries/class-github-get-file-as-html-query.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace RemoteDataBlocks\Example\GitHub;
+
+use RemoteDataBlocks\Config\QueryContext;
+
+class GitHubGetFileAsHtmlQuery extends QueryContext {
+	public array $input_variables = [
+		'file_path' => [
+			'name' => 'File Path',
+			'type' => 'string',
+		],
+		'sha'       => [
+			'name' => 'SHA',
+			'type' => 'string',
+		],
+		'size'      => [
+			'name' => 'Size',
+			'type' => 'number',
+		],
+		'url'       => [
+			'name' => 'URL',
+			'type' => 'string',
+		],
+	];
+
+	public array $output_variables = [
+		'is_collection' => false,
+		'mappings'      => [
+			'file_content' => [
+				'name' => 'File Content',
+				'path' => '$.content',
+				'type' => 'string',
+			],
+			'file_path'    => [
+				'name' => 'File Path',
+				'path' => '$.path',
+				'type' => 'string',
+			],
+			'sha'          => [
+				'name' => 'SHA',
+				'path' => '$.sha',
+				'type' => 'string',
+			],
+			'size'         => [
+				'name' => 'Size',
+				'path' => '$.size',
+				'type' => 'number',
+			],
+			'url'          => [
+				'name' => 'URL',
+				'path' => '$.url',
+				'type' => 'string',
+			],
+		],
+	];
+
+	public function get_endpoint( array $input_variables ): string {
+		return sprintf(
+			'https://api.github.com/repos/%s/%s/contents/%s?ref=%s',
+			$this->get_datasource()->repo_owner,
```

Comment:
Shouldn't this be an input variable?

Reply:
Sounds like yes.

I'll tweak. I need to get my head screwed on better in this part. The way inputs and outputs pipe between each other is still pretty magic to me.